### ISSUE TITLE
Add onAnimationFinishListener for animation

### ISSUE THREAD

**Issue by eresid** [facebook/fresco#181]
Do something if animation done

**Comment by tyronen** [comment]
We don't yet support this.

**Comment by facebook-github-bot-0** [comment]
Thank you for reporting this issue and appreciate your patience. We've  notified the core team for an update on this issue. We're looking for a response within the next 30 days or the issue may be closed.

**Comment by qinchong1988** [comment]
+1

**Comment by cpiz** [comment]
+1

**Comment by guilhermesgb** [comment]
+1

**Comment by plamenko** [comment]
Pull-requests are welcome. Just saying :)

**Comment by dragos-triteanu** [comment]
+1

**Comment by renyanxu2000** [comment]
+1

**Comment by droidekas** [comment]
Has this been picked up?I was considering submitting a PR here,hence the question.

**Comment by oprisnik** [comment]
We're internally working on an improved version that supports this - but it's going to take some time until it is ready. So if you need it sooner feel free to submit a PR.

**Comment by JemliIdea** [comment]
+1

**Comment by k-lam** [comment]
it is ok now?

**Comment by k-lam** [comment]
is it ok now?

**Comment by dragos-triteanu** [comment]
Kewl....thank you guys :D

**Comment by jjding1986** [comment]
+1

**Comment by weisssean** [comment]
+1

**Comment by storix** [comment]
+1

**Comment by k-lam** [comment]
ValueAnimator animator;
new ControllerListener<T>() {
                                @Override
                                public void onSubmit(String id, Object callerContext) {

                                }

                                @Override
                                public void onFinalImageSet(String id, @Nullable ImageInfo imageInfo, @Nullable Animatable animatable) {
                                    animator = ((AnimatedDrawable)animatable).createValueAnimator();
                                }

                                @Override
                                public void onIntermediateImageSet(String id, @Nullable ImageInfo imageInfo) {

                                }

                                @Override
                                public void onIntermediateImageFailed(String id, Throwable throwable) {

                                }

                                @Override
                                public void onFailure(String id, Throwable throwable) {

                                }

                                @Override
                                public void onRelease(String id) {

                                }
                            }

animator is android.animation.ValueAnimator;

**Comment by oprisnik** [comment]
We've released [Fresco v1.4.0](https://github.com/facebook/fresco/releases/tag/v1.4.0), which includes a new implementation for animated GIFs and animated WebPs.

You can now set  an [animation listener](https://github.com/facebook/fresco/blob/master/animated-drawable/src/main/java/com/facebook/fresco/animation/drawable/AnimationListener.java) to get notified about these events.

You can use a `ControllerListener` to attach the `AnimationListener`:
```
Fresco.newDraweeControllerBuilder()
      .setImageRequest(imageRequest)
      .setAutoPlayAnimations(true)
      .setControllerListener(
              new BaseControllerListener() {
                  @Override
                  public void onFinalImageSet(
                          String id, 
                          @Nullable Object imageInfo, 
                          @Nullable Animatable animatable) {
                      if (animatable != null) {
                          AnimatedDrawable2 animatedDrawable = (AnimatedDrawable2) animatable;
                          animatedDrawable.setAnimationListener(yourAnimationListener);
                      }
                  }
              })
      .build()
```
